### ISSUE TITLE
A little trouble with NoVNC

### ISSUE THREAD

**Issue by weeliat** [enkelprifti98/metal-isometric-xepa#3]
Hi, I tried running setup.sh the last couple of days on Equinix metal, however, to my dismay, I was not able to load the VNC remote screen. Upon further investigation, it seems like startxfce4 failed because X was not found in the Rescue mode. Any clue that I can get here to get me moving forward?

**Comment by enkelprifti98** [comment]
Its a known issue but I havent been able to look at it yet.

can you try the alternative based on ubuntu?

https://github.com/enkelprifti98/metal-isometric-xepa-ubuntu

**Comment by zerofai** [comment]
This script work in alpine docker to bring up the noVNC session, although test repo should now be
http://dl-cdn.alpinelinux.org/alpine/edge/testing

But when run the script on Equinix metal,  the gtk3 install return an error, below is output from apk fix

```
apk fix
(1/1) Reinstalling gtk+3.0 (3.24.41-r0)
Executing busybox-1.36.1-r5.trigger
Executing glib-2.76.6-r0.trigger
Executing gtk+3.0-3.24.41-r0.trigger
Error relocating /usr/lib/libgtk-3.so.0: g_assertion_message_cmpint: symbol not found
ERROR: gtk+3.0-3.24.41-r0.trigger: script exited with error 127
OK: 1186 MiB in 540 packages
```

This maybe the reason of why the noVNC does not able to return the desktop, googled a bit but the info related to "g_assertion_message_cmpint" is limited.

**Comment by octopop** [comment]
I was able to get `gtk+3.0` reinstalled that was causing xfce4 to not even operate. However, still failed to get the blank screen fixed.

**Comment by octopop** [comment]
The foundational idea behind this approach seems to be to load an in-memory OS (live CD) and then use libvirt to install a "VM" on physical and boot from it. I wonder if there is a simplification to have folks use netboot.xyz and boot a live Ubuntu or OS of their choice and then follow the libvirt guide detailed here?

NoMachine has much better success than VNC so that may be an option as well for remote desktop; albeit, Alpine is unsupported.

**Comment by enkelprifti98** [comment]
@octopop The ubuntu version of the XEPA project that i linked above is exactly that. It's using the netboot.xyz ubuntu files. I created it because maintaining the alpine version has been a pain as it breaks frequently while the ubuntu version has yet to break over the past 2 years or so.

**Comment by octopop** [comment]
> @octopop The ubuntu version of the XEPA project that i linked above is exactly that. It's using the netboot.xyz ubuntu files. I created it because maintaining the alpine version has been a pain as it breaks frequently while the ubuntu version has yet to break over the past 2 years or so.

Yes that was perfect and really got me and my team bootstrapped!! Thank you for that. Would be awesome if you link the Ubuntu version on your Equinix blog too. Its stellar.

**Comment by enkelprifti98** [comment]
It seems like the issue was that startxfce4 launch script failed to run successfully because gtk+3.0 failed to install. I'm still not sure why that dependency package fails to install but installing gtk+3.0-dev instead works fine.

The script has been updated so it all works now.
